bid-distribution/src/settlement_claims.rs

```diff
@@ -142,28 +128,45 @@ pub fn generate_bid_settlements(
                 "Claims amount is bigger than stakers total claim"
             );
 
-            claims.push(SettlementClaim {
-                withdraw_authority: *settlement_config.marinade_withdraw_authority(),
-                stake_authority: *settlement_config.marinade_stake_authority(),
-                stake_accounts: marinade_fee_deposit_stake_accounts.clone(),
-                claim_amount: marinade_fee_claim,
-                active_stake: marinade_fee_deposit_stake_accounts.values().sum(),
-            });
-            claims_amount += marinade_fee_claim;
-
-            assert!(
-                claims_amount <= effective_total_claim.to_u64().unwrap(),
-                "The sum of total claims exceeds the sum of total staker and marinade fee claims"
-            );
-
-            settlement_claim_collections.push(Settlement {
-                reason: SettlementReason::Bidding,
-                meta: settlement_config.meta().clone(),
-                vote_account: validator.vote_account,
-                claims_count: claims.len(),
-                claims_amount,
-                claims,
-            });
+            let marinade_fee_deposit_stake_accounts: HashMap<_, _> = stake_meta_index
+                .stake_meta_collection
+                .stake_metas
+                .iter()
+                .find(|x| {
+                    x.withdraw_authority
+                        .eq(settlement_config.marinade_withdraw_authority())
+                        && x.stake_authority
+                            .eq(settlement_config.marinade_stake_authority())
+                })
+                .iter()
+                .map(|s| (s.pubkey, s.active_delegation_lamports))
+                .collect();
+
+            if effective_sam_stake > 0 && marinade_fee_claim > 0 {
```

Comment:
Is the `effective_sam_stake` correct here? It's calculated from max stake wanted and not from the stake that is available in the stake accounts (i.e., `active_stake`).
(I would like to understand the meaning of the check)

Reply:
> Is the `effective_sam_stake` correct here? It's calculated from max stake wanted and not from the stake that is available in the stake accounts (i.e., `active_stake`). (I would like to understand the meaning of the check)

my bad it should be `initial_sam_stake` instead. It's not the `total_active_stake` because we should use only the SAM part of the total stake accounts delegated to this validator, since MNDE part is not paying bids.

Reply:
> Is the `effective_sam_stake` correct here? It's calculated from max stake wanted and not from the stake that is available in the stake accounts (i.e., `active_stake`). (I would like to understand the meaning of the check)

sync on discord, please check last changes